crates/iota-indexer/src/handlers/optimistic_pruner.rs

```diff
@@ -0,0 +1,124 @@
+// Copyright (c) 2025 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use std::time::Duration;
+
+use tokio_util::sync::CancellationToken;
+use tracing::{info, warn};
+
+use super::checkpoint_handler::CheckpointHandler;
+use crate::{
+    errors::IndexerError,
+    metrics::IndexerMetrics,
+    store::{IndexerStore, PgIndexerStore, pg_partition_manager::PgPartitionManager},
+    types::IndexerResult,
+};
+
+// Keeping current and previous epoch ensures we will not prune just executed
+// txs on epoch boundary
+const EPOCHS_TO_KEEP: u64 = 2;
+const PRUNING_NOT_IN_PROGRESS_DELAY: Duration = Duration::from_secs(3);
```

Comment:
Nit: I would go with a name  that is more explicit about the triggered operation.
```suggestion
const CHECK_EPOCH_CHANGE_INTERVAL: Duration = Duration::from_secs(3);
```

Reply:
Done in 2b752d8a9241d6c4d5f34a9fc8ee036ed4ecd6e0